src/main/java/com/github/manolo8/darkbot/core/api/GameAPIImpl.java

```diff
@@ -77,14 +79,30 @@ public GameAPIImpl(StartupParams params,
         this.loginData = hasCapability(GameAPI.Capability.LOGIN) ? LoginUtils.performUserLogin(params) : null;
         this.initiallyShown = hasCapability(GameAPI.Capability.INITIALLY_SHOWN) && !params.getAutoHide();
 
+        ConfigAPI config = HeroManager.instance.main.configHandler;
         if (hasCapability(GameAPI.Capability.DIRECT_LIMIT_FPS)) {
-            ConfigAPI config = HeroManager.instance.main.configHandler;
             ConfigSetting<Integer> maxFps = config.requireConfig("bot_settings.api_config.max_fps");
             maxFps.addListener(fpsLimitListener = this::setMaxFps);
             setMaxFps(maxFps.getValue());
         } else {
             this.fpsLimitListener = null;
         }
+
+        if (hasCapability(GameAPI.Capability.PROXY) && (Boolean) config.getConfigValue("bot_settings.api_config.use_proxy")) {
+            new Thread(() -> {
+                int proxy = 7777;
```

Comment:
```suggestion
                int port = 7777;
```
pretty sure port is a better var name here